### PR TITLE
Delete contact + right click menu

### PR DESCRIPTION
**Pull request by BeaudanBrown**
Added ability to delete contacts and destroy all sessions with them
Added right click menu to contacts/conversations in list which can block, delete messages and delete contact
Completes #252

### CODE REVIEW

File: _locales/en/messages.json [hunk 1]
```diff
@@ -953,6 +953,16 @@
     "description":
       "Confirmation dialog text that asks the user if they really wish to delete the conversation. Answer buttons use the strings 'ok' and 'cancel'. The deletion is permanent, i.e. it cannot be undone."
   },
+  "deleteContact": {
```
Comment (Mikunj) [inline _locales/en/messages.json]: Remove this if we don't use it?

File: js/conversation_controller.js [hunk 2]
```diff
@@ -207,6 +212,44 @@
 
       return conversation;
     },
+    async deleteContact(id, type) {
```
Comment (Mikunj) [inline js/conversation_controller.js]: is `type` used anywhere?

File: js/views/conversation_list_item_view.js [hunk 3]
```diff
@@ -26,7 +26,27 @@
...
+      const modelProps = this.model.getPropsForListItem();
+      const props = {
+        ...modelProps,
+        onDeleteContact: () => {
+          Whisper.events.trigger('showConfirmationDialog', {
+            message: i18n('deleteContactConfirmation'),
+            onOk: () =>
+              ConversationController.deleteContact(
+                this.model.id,
+                this.model.get('type')
```
Comment (Mikunj) [inline js/views/conversation_list_item_view.js]: ref: above

File: js/views/conversation_view.js [hunk 4]
```diff
@@ -1447,6 +1448,19 @@
       }
     },
 
+    async deleteContact() {
+      Whisper.events.trigger('showConfirmationDialog', {
+        message: i18n('deleteContactConfirmation'),
+        onOk: () => {
+          ConversationController.deleteContact(
+            this.model.id,
+            this.model.get('type')
```
Comment (Mikunj) [inline js/views/conversation_view.js]: same here

File: js/conversation_controller.js (hunk 2 above)
Comment (BeaudanBrown) [reply js/conversation_controller.js]: Right yeah don't think this is required, just copied from another function

File: _locales/en/messages.json (hunk 1 above)
Comment (BeaudanBrown) [reply _locales/en/messages.json]: This is used as the text in the right click menu and in the cog

**Comment by Mikunj** [comment]
> Deleted contact sends you a message. Is it ignored since it can't be decrypted.

Yep, it should just error out when decrypting and nothing happens

> User decides to re-add the deleted contact.

could you elaborate?

**Comment by sachaaaaa** [comment]
> User decides to re-add the deleted contact.
User A deletes contact B.
User A re-introduces contact B pubkey in the search bar and attempts to send a message.

My guess is that User A will send a friend request message to B.
But I don't think B can handle a friend request in the middle of a session?

**Pull request merged**